src/scripts/ui/inline-roll-links.ts

```diff
@@ -348,9 +348,26 @@ export class InlineRollLinks {
             flags.pf2e.messageId = messageId;
         }
 
-        const actor = resolveActor(foundryDoc ?? resolveDocument(link));
-        if (actor || pf2Traits) {
+        const actor = resolveActor(foundryDoc);
+        const item =
+            foundryDoc instanceof ItemPF2e
+                ? foundryDoc
+                : foundryDoc instanceof ChatMessagePF2e
```

Comment:
Is there a reason you need to check for ChatMessagePF2e? The suggestion I made was to completely omit checking it. The only reason we check for it elsewhere is partly in case there isn't an item, and partly due to legacy. If we do have to check in chat message, that will likely apply to other things such as inline checks, and we would need the resolveItem.

Reply:
Yup, I see it. Line 246 in the current inline-roll-links, inside #onClickInlineCheck. So long as you refactor that one, then you can have resolveItem().

Reply:
As the commit mentioned it was untested, so I wasnt entirely sure what types `foundryDoc` could be at the time.

Having looked at it in Foundry right now, this is indeed not needed.